CMakeLists.txt

```diff
@@ -810,6 +810,7 @@ set(lib_sources ${PROJECT_SOURCE_DIR}/src/ua_types.c
                 ${PROJECT_SOURCE_DIR}/src/pubsub/ua_pubsub_networkmessage.c
                 ${PROJECT_SOURCE_DIR}/src/pubsub/ua_pubsub_writer.c
                 ${PROJECT_SOURCE_DIR}/src/pubsub/ua_pubsub_reader.c
+                ${PROJECT_SOURCE_DIR}/src/pubsub/ua_pubsub_receiver.c
```

Comment:
Writer and Reader are standard-defined concepts.
The code in _receiver clearly belongs to the reader. I would keep it in _reader.c to avoid confusion.
If there is indeed shared functionality with the writer we can have a file for common code and call it such as well.

Reply:
It was my attempt to start a separation of concerns. In the future it might possibly be good to have publisher/subscribe functionality outside of server-coupled code. As the standard describes that server/client and publisher/subscriber are decoupled concepts.